File: pitraining/src/main/java/tn/esprit/pitraining/services/QuizQuestionService.java
package tn.esprit.pitraining.services;

import tn.esprit.pitraining.entities.QuizQuestion;

import java.util.List;
import java.util.Optional;

public interface QuizQuestionService {

    List<QuizQuestion> getQuestionsByQuizId(Long quizId);

    Optional<QuizQuestion> getQuestionById(Long quizId, Long id);

    QuizQuestion saveQuestion(Long quizId, QuizQuestion question);

    QuizQuestion updateQuestion(Long quizId, Long id, QuizQuestion updatedQuestion);

    void deleteQuestion(Long quizId, Long id);
}

File: 2104 - 2/front/src/app/views/tmfiles/services/quiz-question.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { QuizQuestion } from '../quiz/quiz-question.model'; // Your QuizQuestion model

@Injectable({ providedIn: 'root' })
export class QuizQuestionService {
  private backendUrl = 'http://localhost:8080/api';

  constructor(private http: HttpClient) {}

  getQuestionsByQuizId(quizId: number): Observable<QuizQuestion[]> {
    return this.http.get<QuizQuestion[]>(`${this.backendUrl}/quizzes/${quizId}/questions`);
  }

  getQuestionById(quizId: number, questionId: number): Observable<QuizQuestion> {
    return this.http.get<QuizQuestion>(`${this.backendUrl}/quizzes/${quizId}/questions/${questionId}`);
  }

  createQuestion(quizId: number, newQuestion: QuizQuestion): Observable<QuizQuestion> {
    return this.http.post<QuizQuestion>(`${this.backendUrl}/quizzes/${quizId}/questions`, newQuestion);
  }

  updateQuestion(quizId: number, questionId: number, updatedQuestion: QuizQuestion): Observable<QuizQuestion> { 
    return this.http.put<QuizQuestion>(`${this.backendUrl}/quizzes/${quizId}/questions/${questionId}`, updatedQuestion);
  }

  deleteQuestion(quizId: number, questionId: number): Observable<void> { 
    return this.http.delete<void>(`${this.backendUrl}/quizzes/${quizId}/questions/${questionId}`);
  }
}
